### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by nnabse**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -15,6 +15,8 @@ import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
 import { MatTableModule } from "@angular/material/table";
 import { MatProgressSpinnerModule } from "@angular/material/progress-spinner";
 import { AbsolutePipe } from './pipes/absolute.pipe';
```
Comment (MkpDolganovKirill) [inline src/app/app.module.ts]: import ordering

File: src/app/app.module.ts [hunk 2]
```diff
@@ -35,6 +37,8 @@ import { AbsolutePipe } from './pipes/absolute.pipe';
     MatTableModule,
     BrowserAnimationsModule,
     MatProgressSpinnerModule,
+    MatAutocompleteModule,
```
Comment (MkpDolganovKirill) [inline src/app/app.module.ts]: Mat components need to switch to other file

File: src/app/components/left-side-bar/left-side-bar.component.scss [hunk 3]
```diff
@@ -1,4 +1,4 @@
-@import 'src/variables';
+@import "src/variables";
```
Comment (MkpDolganovKirill) [inline src/app/components/left-side-bar/left-side-bar.component.scss]: variables

File: src/app/pages/crypto-rates/crypto-rates.component.html [hunk 4]
```diff
@@ -1,55 +1,142 @@
 <div class="rates-container">
+  <form class="example-form">
+    <mat-form-field class="example-full-width" appearance="fill">
+      <mat-label>{{ searchField.PLACEHOLDER }}</mat-label>
+      <input matInput aria-label="State" [matAutocomplete]="auto" />
+      <mat-autocomplete #auto="matAutocomplete">
+        <mat-option *ngFor="let item of cryptoList" [value]="item.name">
+          <img
+            src="https://s2.coinmarketcap.com/static/img/coins/64x64/{{
```
Comment (MkpDolganovKirill) [inline src/app/pages/crypto-rates/crypto-rates.component.html]: magic string

File: src/app/pages/crypto-rates/crypto-rates.component.html [hunk 5]
```diff
@@ -1,55 +1,142 @@
...
-  <mat-spinner *ngIf="!data?.length"></mat-spinner>
   <div class="table-container">
-    <table *ngIf="data?.length" mat-table [dataSource]="data" class="mat-elevation-z8">
-
+    <div *ngIf="!data.data.length" class="spinner">
+      <mat-spinner></mat-spinner>
+    </div>
+    <table mat-table [dataSource]="data" class="mat-elevation-z8">
       <ng-container matColumnDef="number">
-        <th mat-header-cell *matHeaderCellDef> No. </th>
```
Comment (MkpDolganovKirill) [inline src/app/pages/crypto-rates/crypto-rates.component.html:8]: magic string

File: src/app/pages/crypto-rates/crypto-rates.component.html [hunk 6]
```diff
@@ -1,55 +1,142 @@
...
-        <th mat-header-cell *matHeaderCellDef> Логотип </th>
+      <ng-container matColumnDef="name">
+        <th mat-header-cell *matHeaderCellDef class="label">
+          {{ tableTitle.NAME }}
+        </th>
         <td mat-cell *matCellDef="let element">
-          <img src="https://s2.coinmarketcap.com/static/img/coins/64x64/{{element.cap_id}}.png" alt="logo">
+          <div class="content">
+            <img
+              src="https://s2.coinmarketcap.com/static/img/coins/64x64/{{
```
Comment (MkpDolganovKirill) [inline src/app/pages/crypto-rates/crypto-rates.component.html]: magic string

File: src/app/pages/crypto-rates/crypto-rates.component.scss [hunk 7]
```diff
@@ -1,30 +1,92 @@
+@import "variables";
+
 .rates-container {
   width: 100%;
   max-height: 100%;
   display: flex;
   flex-direction: column;
-  gap: 5px;
+  justify-content: center;
+  padding: 30px;
```
Comment (MkpDolganovKirill) [inline src/app/pages/crypto-rates/crypto-rates.component.scss:9]: why did you change gap on padding?

File: src/app/pages/crypto-rates/crypto-rates.enums.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+enum TableColor {
+  RED = 'red',
+  GREEN = 'green',
+}
+
+enum TableStyle {
+  ROTATE = 'rotate',
+}
+
+enum TableLink {
```
Comment (MkpDolganovKirill) [inline src/app/pages/crypto-rates/crypto-rates.enums.ts]: enum for one link? use constants

File: src/app/services/api.service.ts [hunk 9]
```diff
@@ -1,21 +1,22 @@
...
   getCryptoInfo() {
-    this.httpClient.get(`/crypto/top`, {
-     params: {
-       limit: 100
-     }
-    })
+    this.httpClient
+      .get<CryptoItem[]>(`${environment.backendLink}/crypto/top`, {
+        params: {
+          limit: 1000,
```
Comment (MkpDolganovKirill) [inline src/app/services/api.service.ts]: limit in constants or use this for change in UI

File: src/variables.scss [hunk 10]
```diff
@@ -1,5 +1,5 @@
 $primary-color: #2a2f23;
-$secondary-color: #AFBBF2;
-$third-color: #AFBBF2;
-$text-color-light: #D3FFF3;
+$secondary-color: #afbbf2;
```
Comment (MkpDolganovKirill) [inline src/variables.scss:2]: two same variables

**Review by MkpDolganovKirill — CHANGES_REQUESTED** [review]

File: src/app/pages/crypto-rates/crypto-rates.component.scss (hunk 7 above)
Comment (nnabse) [reply src/app/pages/crypto-rates/crypto-rates.component.scss:9]: its ur commit

**Review by MkpDolganovKirill — APPROVED** [review]

**Pull request merged**